### ISSUE TITLE
Uncaught a reference to the main browser is required by this constructor

### ISSUE THREAD

**Issue by lipan6461188** [GMOD/jbrowse#1488]
# Pre-requisites

JBrowser (v1.16.8)
Embed in <iframe>
Web browser: Chrome
Operating system: Windows 10

```
<iframe src="" style="border: 1px solid black; width:100%; height:100vh;" id="jbrowser">
</iframe>

<script>
function whenEvent(){
	var link = "...";
	$('#jbrowser').attr('src', link);
}
</script>
```

I highlight a gene, and click [Save track data] on a BigWig track. But it show nothing. The error occured in console.

Component.js:18 Uncaught a reference to the main browser is required by this constructor

registry.js:26 Uncaught Error: Tried to register widget with id==exportDialog but that id is already registered
    at Object.add (registry.js:26)
    at Object.create (_WidgetBase.js:428)
    at Object.inherited (declare.js:214)
    at Object.create (ContentPane.js:174)
    at Object.postscript (_WidgetBase.js:366)
    at new eval (declare.js:432)
    at Object._openDialog (BlockBased.js:881)
    at Object.handler [as onClick] (BlockBased.js:691)
    at Object.onItemClick (_MenuBase.js:318)
    at HTMLTableRowElement.eval (_MenuBase.js:101)

[Image](http://rasp.zhanglab.net/static/upload_imgs/issue.png)

**Comment by cmdcolin** [comment]
This is a total guess but you might want to fully destroy and recreate the iframe

**Comment by lipan6461188** [comment]
> This is a total guess but you might want to fully destroy and recreate the iframe

It doesn't work. Still got the same error

```
function whenEvent(){
    $('#iframe_container').empty();
    var link = "...";
    var iframe = $('<iframe src='+link+'></iframe>');
    iframe.attr('style', "border: 1px solid black; width:100%; height:100vh;")
    iframe.attr('id', "jbrowser")
    $('#iframe_container').append(iframe)
}
```

**Comment by cmdcolin** [comment]
I don't see any such error with your original code actually

```
<html>
   <head>
      <script src="https://cdn.jsdelivr.net/npm/jquery@3.4.1/dist/jquery.min.js"></script>
   </head>
   <body>
      <iframe src="" style="border: 1px solid black; width:100%; height:100vh;" id="jbrowser">
      </iframe>
      <button onClick="whenEvent()">Test</button>
      <script>
         function whenEvent(){
                 var link = "http://localhost/jbrowse/?data=sample_data/json/volvox"
                 $('#jbrowser').attr('src', link);
         }
      </script>
   </body>
</html>
```

This works fine after a save track data and after clicking the button multiple times

**Comment by lipan6461188** [comment]
> I don't see any such error with your original code actually
> 
> ```
> <html>
>    <head>
>       <script src="https://cdn.jsdelivr.net/npm/jquery@3.4.1/dist/jquery.min.js"></script>
>    </head>
>    <body>
>       <iframe src="" style="border: 1px solid black; width:100%; height:100vh;" id="jbrowser">
>       </iframe>
>       <button onClick="whenEvent()">Test</button>
>       <script>
>          function whenEvent(){
>                  var link = "http://localhost/jbrowse/?data=sample_data/json/volvox"
>                  $('#jbrowser').attr('src', link);
>          }
>       </script>
>    </body>
> </html>
> ```
> 
> This works fine after a save track data and after clicking the button multiple times

Thank you so much. Could you please try this URL:
```
<html>
   <head>
      <script src="https://cdn.jsdelivr.net/npm/jquery@3.4.1/dist/jquery.min.js"></script>
   </head>
   <body>
      <iframe src="" style="border: 1px solid black; width:100%; height:100vh;" id="jbrowser">
      </iframe>
      <button onClick="whenEvent()">Test</button>
      <script>
         function whenEvent(){
                 var link = "http://jbrowser.zhanglab.net?data=RASP/hg38&loc=chr12:6533037..6539264&tracks=refseq_genes,DMS_Genome-wide_fbl_vivo-pos_score"
                 $('#jbrowser').attr('src', link);
         }
      </script>
   </body>
</html>
```

Highlight the GAPDH gene and click [Save track data] of "DMS_Genome-wide_fbl_vivo-pos_score" track. It seems doesn't work.

**Comment by cmdcolin** [comment]
Thanks this is reproducible now, looks like there is something related to doing this highlight and then doing a Save track data perhaps, or this at least makes it reproducible

**Comment by cmdcolin** [comment]
Possible fix in #1489 

We can probably get this deployed in the next release, or if you want you can try to test this branch

Thanks for reporting!
